ggplot lesson revisions (#16)
bcgov/ds-intro-to-r-2-day: 06-plot-ggplot2.Rmd

> >
> > ```{r ch4a-sol}
> > ggplot(data = gapminder, mapping = aes(x = gdpPercap, y = lifeExp)) +
> >  geom_point(size = 3, colour = "orange") +

> > > scale_x_log10() +

Reviewer: ```suggestion
> >  scale_x_log10() +
```

ggplot(data = asia_oceania, mapping = aes(x = year, y = lifeExp)) +
  geom_line() + 
  facet_wrap(vars(country))
```

> The `facet_wrap` layer took the function `vars()` as its argument. This tells R to draw a panel for each unique value in the variable country of the gapminder dataset.

Reviewer: ```suggestion
Here we supplied the `vars` function to `facet_wrap`'s `facets` argument to easily specify which variable(s) we want to create panels for. In this case, we are telling R to draw a panel for each unique value in the variable country of the gapminder dataset.
```

Author: Thanks for the the suggested edit, I think I am going to make a mash-up of this with the original commit.

```{r theme}
ggplot(data = asia_oceania, mapping = aes(x = year, y = lifeExp, colour = continent)) +
  geom_line() +
  facet_wrap(vars(country)) +
  labs(
    x = "Year",              # x axis title
    y = "Life expectancy",   # y axis title
    title = "Figure 1",      # main title of figure
    colour = "Continent"      # title of legend
  ) +
  theme(axis.text.x = element_text(size = 5),
        plot.title = element_text(face = "bold"))
```

> ## Tip

> > Note that we apply a "theme" definition to customaize 

Reviewer: ```suggestion
> Note that we apply a "theme" definition to customize 
```

> the x-axis label sizes to maintain readability.  Nearly everything in

> > ggplot2 is customizable.

Reviewer 2: Great addition showing the built-in themes. What do you think about a quick demo of `?theme` and `?ggtheme` to show the miriad of options and reinforce using `?` help?

Author: Yeah, I think that is a great idea. Do you think this needs to be captured in the bookdown or could it be an 'on the fly' element?
